tests: ethernet: tests/drivers/build_all/ethernet : fails on frdm_k64f

### Describe the bug

- frdm_k64f

** git bisect

1fd4886ad97c185a9b5ad081a92075b6f2c7891a is the first bad commit
```
commit 1fd4886ad97c185a9b5ad081a92075b6f2c7891a
Author: Tony Han <tony.han@microchip.com>
Date:   Tue May 27 15:44:30 2025 +0800

    drivers: ethernet: phy: ksz8081: add support for interrupt mode
    
    Enable Link-Up and Link-Down interrupts. On the interrupt handling
    the monitor work is scheduled to update the link status and calling
    corresponding callback routine.
    
    Signed-off-by: Tony Han <tony.han@microchip.com>
```

### Regression

- [x] This is a regression.

### Steps to reproduce

```
west build -b frdm_k64f tests/drivers/build_all/ethernet
```

### Relevant log output

```shell
/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET' 64 | __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code) | ^~~~~~~~~~~~~~~~~~~~/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE' 59 | __COND_CODE(_XXXX##_flag, _if_1_code, _else_code) | ^~~~~~~~~~~/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/include/zephyr/sys/util_macro.h:427:24: note: in expansion of macro 'COND_CODE_1' 427 | #define UTIL_AND(a, b) COND_CODE_1(UTIL_BOOL(a), (b), (0)) | ^~~~~~~~~~~/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/drivers/ethernet/phy/phy_microchip_ksz8081.c:67:25: note: in expansion of macro 'UTIL_AND' 67 | UTIL_AND(DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios), \ | ^~~~~~~~/home/jenkins/agent/workspace/zephyr_downstream_build_daily/zephyr/drivers/ethernet/phy/phy_microchip_ksz8081.c:514:13: note: in expansion of macro 'USING_INTERRUPT_GPIO' 514 | if (USING_INTERRUPT_GPIO) { | ^~~~~~~~~~~~~~~~~~~~ninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: ubuntu
- tool-chain: 0.17.1
- version: 8f85d8e62ac5bec45e703086ee28583e9f67baac

### Additional Context

_No response_

--- a/drivers/ethernet/phy/phy_microchip_ksz8081.c
+++ b/drivers/ethernet/phy/phy_microchip_ksz8081.c
@@ -474,6 +474,7 @@ done:
 static int phy_mc_ksz8081_cfg_link(const struct device *dev, enum phy_link_speed speeds,
 				   enum phy_cfg_link_flag flags)
 {
+	__maybe_unused const struct mc_ksz8081_config *config = dev->config;
 	struct mc_ksz8081_data *data = dev->data;
 	int ret;
 

